packages/attestation-service/src/sms/twilio.ts

```diff
@@ -76,15 +119,76 @@ export class TwilioSmsProvider extends SmsProvider {
     } catch (error) {
       throw new Error(`Twilio Messaging Service could not be fetched: ${error}`)
     }
+    if (this.verifyServiceSid) {
+      try {
+        await this.client.verify.services
+          .get(this.verifyServiceSid)
+          .fetch()
+          .then((service) => {
+            if (!service.customCodeEnabled) {
+              // Make sure that custom code is enabled
+              throw new Error(
+                'TWILIO_VERIFY_SERVICE_SID is specified, but customCode is not enabled. Please contact Twilio support to enable it.'
+              )
+            }
+          })
+      } catch (error) {
+        throw new Error(`Twilio Verify Service could not be fetched: ${error}`)
+      }
+    }
   }
 
   async sendSms(attestation: AttestationModel) {
-    const m = await this.client.messages.create({
-      body: attestation.message,
-      to: attestation.phoneNumber,
-      from: this.messagingServiceSid,
-      statusCallback: this.deliveryStatusURL,
-    })
-    return m.sid
+    // Prefer Verify API if Verify Service is present and
+    // user is not requesting a deeplink
+    if (this.verifyServiceSid && attestation.securityCode) {
+      const requestParams: any = {
+        to: attestation.message,
```

Comment:
doesn't this message also include the securityCode?

` messageBase = ${getSecurityCodeText(this.attestationRequest.language)}: ${securityCode}`

i'm making the assumption that Twilio appends the customCode in the SMS, but maybe that's not the case?

Reply:
This should be `to: attestaion.phoneNumber`
I'm not sure where this came from 😕

Reply:
I see to the verifyAPI constructs the message completely itself (including appending the <#> prefix and the apphash). So we just need to pass the code we want.

But ya, how does this work if we are sending the message in the `to` field?